src/sparseml/pytorch/utils/quantization/helpers.py

```diff
@@ -64,6 +67,214 @@
+            constructor from the attributes of the given Module
+        """
+        qat_wrapper_param_names = [
+            "num_inputs",
+            "num_outputs",
+            "input_qconfigs",
+            "output_qconfigs",
+        ]
+        qat_wrapper_kwargs = {
+            param_name: getattr(module, param_name)
+            for param_name in qat_wrapper_param_names
+            if hasattr(module, param_name)
+        }
+
+        return QATWrapper(forward_fn=module, **qat_wrapper_kwargs)
+
+    def __init__(
+        self,
+        forward_fn: Callable[[Any], Any],
+        num_inputs: int = 1,
+        num_outputs: int = 1,
+        input_qconfigs: Union[
+            "torch.quantization.QConfig", str, List["torch.quantization.QConfig"]
+        ] = "asymmetric",
+        output_qconfigs: Union[
+            "torch.quantization.QConfig", str, List["torch.quantization.QConfig"]
+        ] = "asymmetric",
+    ):
+        super().__init__()
+
+        if torch_quantization is None:
+            raise RuntimeError(
+                "Unable to import package torch.quantization. "
+                "Try upgrading your PyTorch version."
+            )
+
+        if not callable(forward_fn):
+            raise ValueError(
+                "forward_fn of QATWrapper must be callable. "
+                f"Received {type(forward_fn)}"
+            )
+
+        self.forward_fn = forward_fn
+        self.input_qconfigs = self._load_qconfigs(
+            "input_qconfigs", num_inputs, input_qconfigs
+        )
+        self.output_qconfigs = self._load_qconfigs(
+            "output_qconfigs", num_outputs, output_qconfigs
+        )
+
+        self.input_quant_stubs = torch.nn.ModuleList(
+            [torch_quantization.QuantStub() for _ in range(num_inputs)]
+        )
+        self.output_quant_stubs = torch.nn.ModuleList(
+            [torch_quantization.QuantStub() for _ in range(num_outputs)]
+        )
+        self.output_dequant_stubs = torch.nn.ModuleList(
+            [torch_quantization.DeQuantStub() for _ in range(num_outputs)]
+        )
+
+    def forward(self, *args, **kwargs) -> Any:
+        """
+        :param args: arguments to forward function; the first num_inputs of these args
+            will be wrapped by a QuantStub
+        :param kwargs: key word arguments to pass to the wrapped forward function
+        :return: outputs of the forward function with a QuantStub applied to the first
+            num_outputs outputs
+        """
+
+        qat_args = []
+
+        for idx, arg in enumerate(args):
+            if idx < len(self.input_quant_stubs):
+                arg = self.input_quant_stubs[idx](arg)
+            qat_args.append(arg)
+
+        outputs = self.forward_fn(*qat_args, **kwargs)
+
+        if len(self.output_quant_stubs) == 0:
+            # no output wrapping
+            return outputs
+
+        if isinstance(outputs, torch.Tensor):
```

Comment:
could we also check for this case that the setup was told to only create one output_quant_stub and otherwise raise an exception?